enterprise/frontend/src/embedding-sdk/lib/interactive-question/update-question.ts

```diff
@@ -43,19 +48,30 @@ export const runQuestionOnQueryChangeSdk =
       shouldStartAdHocQuestion,
     });
 
+    if (!nextQuestion.canAutoRun()) {
+      run = false;
+    }
+
     const rawSeries = createRawSeries({
       question: nextQuestion,
       queryResult: queryResults?.[0],
       datasetQuery: undefined,
     });
 
-    const questionPivotResult = computeQuestionPivotTable({
+    const computedPivotQuestion = computeQuestionPivotTable({
       question: nextQuestion,
       currentQuestion: previousQuestion,
       rawSeries,
     });
 
-    nextQuestion = questionPivotResult.question;
+    nextQuestion = computedPivotQuestion.question;
+
+    if (computedPivotQuestion.shouldRun !== null) {
+      run = computedPivotQuestion.shouldRun;
```

Comment:
can we calculate `run` in some sort of a one-liner or function? I think it would be easier to read and maintain if someone else looks into this code in the future

Reply:
@oisincoveney we do have `checkShouldRerunPivotTableQuestion(...)` inside of the method, but the issue is that relies on other parameters as seen in `computeQuestionPivotTable`

```ts
export function computeQuestionPivotTable(options: Options) {
  const { currentQuestion, rawSeries } = options;

  let { question } = options;

  const isPivot = question.display() === "pivot";
  const wasPivot = currentQuestion?.display() === "pivot";
  let shouldRun = null;

  const isQuestionNative = Lib.queryDisplayInfo(question.query()).isNative;

  if (wasPivot || isPivot) {
    const hasBreakouts =
      !isQuestionNative && Lib.breakouts(question.query(), -1).length > 0;

    // compute the pivot setting now so we can query the appropriate data
    if (isPivot && hasBreakouts) {
      const key = "pivot_table.column_split";

      if (rawSeries) {
        const series = assocIn(rawSeries, [0, "card"], question.card());
        const setting = getQuestionWithDefaultVisualizationSettings(
          question,
          series,
        ).setting(key);

        question = question.updateSettings({ [key]: setting });
      }
    }

    shouldRun = checkShouldRerunPivotTableQuestion({
      isPivot,
      wasPivot,
      hasBreakouts,
      currentQuestion,
      question,
    });
  }

  return { shouldRun, question };
}
```

Reply:
ok! no worries then. was wondering if there was a way to simplify that